polaris-react/src/components/DataTable/DataTable.tsx

```diff
@@ -438,7 +441,7 @@ class DataTableInner extends PureComponent<CombinedProps, DataTableState> {
       button.addEventListener('focus', this.handleHeaderButtonFocus);
     } else {
       this.tableHeadings[index] = ref;
-      this.tableHeadingWidths[index] = ref.getBoundingClientRect().width;
+      this.tableHeadingWidths[index] = ref.clientWidth;
```

Comment:
Technically `getBoundingClientRect().width` gives a more accurate number, but the rounding ends up being inconsistent between the headings and the headings in the sticky header table. Overall `clientWidth` gives more consistent results as it provides a integer vs a float.